### ISSUE TITLE
[0.11.1] snapshot request to meta node timeout intermittently

### ISSUE THREAD

**Issue by supershal** [influxdata/influxdb#6408]
### Bug report
influxdb version: 0.11.1
Cluster: 3 meta node and 3 data nodes
The data node timout while getting snapshot from meta server.
```
[metaclient] 2016/04/18 01:58:25 failure getting snapshot from <MetaNode>:8091: meta server returned non-200: 504 GATEWAY_TIMEOUT
```
I got tcp dump on the call. The request was: ```<MetaNode1>:8091?index=444```.
I logged to the meta node and made same request locally. ```curl -v `hostname`:8091?index=444``` . The call hands indefinitely. 
So I restarted the meta server and made the same request, it succeeded. However the next request on index=445 failed. Restarting meta node again makes snapshot with 445 index available but next one (446) is unavailable. I dont see any error in meta server's logs. This repeats.

I did some debugging: I think the db store is not getting next snapshot index and hangs indefinitely. serveSnapshot() method at https://github.com/influxdata/influxdb/blob/0.11/services/meta/handler.go
I will debug more, but I will need help understanding the code.
``` 
select {
	case <-h.store.afterIndex(index):
		// Send updated snapshot to client.
		ss, err := h.store.snapshot()
		if err != nil {
			h.httpError(err, w, http.StatusInternalServerError)
			return
		}
		b, err := ss.MarshalBinary()
		if err != nil {
			h.httpError(err, w, http.StatusInternalServerError)
			return
		}
		w.Header().Add("Content-Type", "application/octet-stream")
		w.Write(b)
		return
	case <-w.(http.CloseNotifier).CloseNotify():
		// Client closed the connection so we're done.
		return
	case <-h.closing:
		h.httpError(fmt.Errorf("server closed"), w, http.StatusInternalServerError)
		return
```

Please let me know if there is any workaround meanwhile.

**Comment by jwilder** [comment]
Are you using a load balancer in front of your meta nodes?  `504 GATEWAY_TIMEOUT` doesn't look like an error returned by influxdb.  

The endpoint your calling directly is an internal endpoint used by the cluster.  It will block and stay open indefinitely until the index requested is available so the behavior your describe looks correct.  If there is a load balancer or proxy in front of that call, it may time out the connection though.

Please keep in mind that clustering code has been removed in the 0.12 release.

**Comment by supershal** [comment]
Thanks @jwilder . I have ELB in front of meta nodes. I did increased elb timeouts first to make sure elb is functioning properly or not.
I executed the request locally on meta node and noticed that the request is hanging indefinitely (reason for ELB timeout). However the indefinite blocking of the request threw me off.

As an API user I would expect 404 to be returned when index is not available. 
This explains why it was working intermittently. 

I understand clustering code is removed in 0.12.
